### PR TITLE
Additional Microsoft Modules to track in PS7 Telemetry

### PR DESCRIPTION
**Pull request by SydneyhSmith**
# PR Summary

This adds additional Microsoft-owned modules to be tracked at load time in the PowerShell 7 telemetry. 

## PR Context

This allows us to better understand how PowerShell is being used to manage services and to better understand the types of PowerShell users based on which modules are loaded into PowerShell sessions. 

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [X] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [X] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/utils/Telemetry.cs [hunk 1]
```diff
@@ -112,17 +112,247 @@ static ApplicationInsightsTelemetry()
...
+                        "nPSDesiredStateConfiguration",
+                        "nPSDesiredStateConfiguration",
+                        "OperationsManager",
                         "PackageManagement",
+                        "PartnerCenter",
                         "Pester",
+                        "PKITools",
+                        "PoShGroupPolicy",
+                        "PoshPctBar",
+                        "PoshWSUS",
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/utils/Telemetry.cs]: This is not a Msft module and I suspect others in this are also not.  Please rereview.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/utils/Telemetry.cs (hunk 1 above)
Comment (SydneyhSmith) [reply src/System.Management.Automation/utils/Telemetry.cs]: You are correct, re-did the list and manually verified that all the remaining modules are MSFT owned

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]
@joeyaiello needs to review this list for accuracy

**Comment by TravisEz13** [comment]
@SteveL-MSFT does this need to get in the RC?

**Comment by SteveL-MSFT** [comment]
@TravisEz13 yes

File: src/System.Management.Automation/utils/Telemetry.cs [hunk 2]
```diff
@@ -112,17 +112,431 @@ static ApplicationInsightsTelemetry()
...
+                        "shieldedvmprovisioning",
+                        "shieldedvmtemplate",
+                        "skype",
+                        "SkypeForBusinessHybridHealth",
+                        "smbshare",
+                        "smbwitness",
+                        "smisconfig",
+                        "softwareinventorylogging",
+                        "SPFAdmin",
+                        "spmt",
```
Comment (joeyaiello) [inline src/System.Management.Automation/utils/Telemetry.cs]:
```suggestion
                        "Microsoft.SharePoint.MigrationTool.PowerShell",
```

File: src/System.Management.Automation/utils/Telemetry.cs [hunk 3]
```diff
@@ -112,17 +112,431 @@ static ApplicationInsightsTelemetry()
...
+                        "sqlps",
+                        "SqlServer",
+                        "staffhub",
```
Comment (joeyaiello) [inline src/System.Management.Automation/utils/Telemetry.cs]: Dupe of MicrosoftStaffHub
```suggestion
```

File: src/System.Management.Automation/utils/Telemetry.cs [hunk 4]
```diff
@@ -112,17 +112,431 @@ static ApplicationInsightsTelemetry()
...
+                        "startlayout",
+                        "startscreen",
+                        "storage",
+                        "StorageDsc",
+                        "storageqos",
+                        "storagereplica",
+                        "storagespaces",
+                        "syncshare",
+                        "System.Center.Service.Manager",
+                        "teams",
```
Comment (joeyaiello) [inline src/System.Management.Automation/utils/Telemetry.cs]: Dupe of MicrsosoftTeams

```suggestion
```

File: src/System.Management.Automation/utils/Telemetry.cs [hunk 5]
```diff
@@ -112,17 +112,431 @@ static ApplicationInsightsTelemetry()
...
+                        "tls",
+                        "troubleshootingpack",
+                        "trustedplatformmodule",
+                        "uev",
+                        "updateservices",
+                        "useraccesslogging",
+                        "vamt",
+                        "VirtualMachineManager",
+                        "vpnclient",
+                        "waspsext",
```
Comment (joeyaiello) [inline src/System.Management.Automation/utils/Telemetry.cs]:
```suggestion
                        "WasPSExt",
```

File: src/System.Management.Automation/utils/Telemetry.cs [hunk 6]
```diff
@@ -112,17 +112,431 @@ static ApplicationInsightsTelemetry()
...
+                        "wdac",
+                        "wds",
+                        "webadministration",
+                        "WebAdministrationDsc",
+                        "webadminstration",
+                        "webapplicationproxy",
+                        "WebSites",
+                        "whea",
+                        "whiteboard",
```
Comment (joeyaiello) [inline src/System.Management.Automation/utils/Telemetry.cs]:
```suggestion
                        "WhiteboardAdmin",
```

File: src/System.Management.Automation/utils/Telemetry.cs [hunk 7]
```diff
@@ -112,17 +112,431 @@ static ApplicationInsightsTelemetry()
...
+                        "WindowsDefender",
+                        "WindowsDefenderDsc",
+                        "windowsdeveloperlicense",
+                        "windowsdevlic",
```
Comment (joeyaiello) [inline src/System.Management.Automation/utils/Telemetry.cs]: Dupe of the one above it
```suggestion
```

**Review by joeyaiello — CHANGES_REQUESTED** [review]
Whew. Finally got through this one. 

We have lots of work to do in renaming Windows PowerShell modules in the Microsoft Docs, @sdwheeler, I took some notes along the way.

**Review by SteveL-MSFT — COMMENTED** [review]
Ok from me once @joeyaiello approves

**Comment by adityapatwardhan** [comment]
@SteveL-MSFT @joeyaiello  Please re-review.

**Review by joeyaiello — APPROVED** [review]
Thanks for the changes. Approving operating under the assumption that this list is not case sensitive (assuming that's what `StringComparer.OrdinalIgnoreCase` does, but I never can remember the exact behavior around ordinal vs. non-ordinal). 

If it is case-sensitive, I have to make some more changes ASAP

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by troymoore** [comment]
Everythings good dude

**Pull request merged**